community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/spi/InstrumentedGraphStatistics.scala

```diff
@@ -48,37 +48,15 @@ case class GraphStatisticsSnapshot(map: Map[StatisticsKey, Double] = Map.empty)
     snapshot.freeze
   }
 
+  //A plan has diverged if there is a relative change in any of the
+  //statistics that is bigger than the threshold
   def diverges(snapshot: GraphStatisticsSnapshot, minThreshold: Double): Boolean = {
     assert(map.keySet == snapshot.map.keySet)
-    def vectorLength(map: Map[StatisticsKey, Double]): Double =
-      Math.sqrt(map.values.map(Math.pow(_, 2)).kahanSum)
-
-    val dotProduct: Double = (map.toSeq ++ snapshot.map.toSeq)
```

Comment:
`vectorLength` and `dotProduct` here treat selectivities and cardinalities equally, meaning that changes in selectivities will be completely dwarfed by changes in cardinalities.